### PR TITLE
feat/prepending txs

### PR DESCRIPTION
**Pull request by mroz22**

### CODE REVIEW

File: packages/connect/src/api/bitcoin/createPending.ts [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        value: valueOut.toString(),
+        valueIn: valueIn.toString(),
+        fees: valueIn.minus(valueOut).toString(),
+        vin: inputs.map((ins, n) => ({
+            n,
+            txid: ins.prev_hash,
+            vout: ins.prev_index,
+            isAddress: true,
+            addresses: utxo!
+                .filter(utxo => utxo.txid + utxo.vout === ins.prev_hash + ins.prev_index)
```
Comment (szymonlesisz) [inline packages/connect/src/api/bitcoin/createPending.ts]: utxo.txid === ins.prev_hash && utxo.vout === ins.prev_index

File: packages/connect/src/api/bitcoin/index.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+export * from './createPending';
```
Comment (szymonlesisz) [inline packages/connect/src/api/bitcoin/index.ts]: createPendingTx (like in coinjoin package) ?

File: packages/connect/src/api/signTransaction.ts [hunk 3]
```diff
@@ -233,16 +238,32 @@ export default class SignTransaction extends AbstractMethod<'signTransaction', P
...
             if (bitcoinTx.hasWitnesses()) {
                 response.witnesses = bitcoinTx.ins.map((_, i) =>
-                    bitcoinTx.getWitness(i)?.toString('hex'),
+                    bitcoinTx?.getWitness(i)?.toString('hex'),
                 );
             }
         }
 
+        if (bitcoinTx && params.addresses && params.utxo) {
+            response.pendingTransaction = {
```
Comment (szymonlesisz) [inline packages/connect/src/api/signTransaction.ts]: signedTransaction?

File: packages/connect/src/api/bitcoin/createPending.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    const valueOut = outputs.reduce(
+        (sum, out) => BigNumber(sum).plus(out.amount),
+        new BigNumber('0'),
+    );
+    const valueIn = inputs.reduce(
+        (sum, ins) => BigNumber(sum).plus(ins.amount),
+        new BigNumber('0'),
+    );
+    return {
+        hex: tx.toHex(),
```
Comment (szymonlesisz) [inline packages/connect/src/api/bitcoin/createPending.ts]: id: tx.getId()

File: packages/connect/src/api/signTransaction.ts [hunk 5]
```diff
@@ -233,16 +238,32 @@ export default class SignTransaction extends AbstractMethod<'signTransaction', P
...
+        }
+
         if (params.push) {
             // validate backend
             isBackendSupported(params.coinInfo);
             const blockchain = await initBlockchain(params.coinInfo, this.postMessage);
             const txid = await blockchain.pushTransaction(response.serializedTx);
+
+            if (response.pendingTransaction) {
+                response.pendingTransaction.txid = txid;
```
Comment (szymonlesisz) [inline packages/connect/src/api/signTransaction.ts]: from tx.getId()

File: packages/suite/src/actions/wallet/sendFormActions.ts [hunk 6]
```diff
@@ -208,121 +210,152 @@ export const cancelSignTx = () => (dispatch: Dispatch, getState: GetState) => {
...
+                    symbol: account.symbol,
+                    txid,
+                }),
+            );
+
+            if (precomposedTx.prevTxid) {
+                // notification from the backend may be delayed.
+                // modify affected transaction(s) in the reducer until the real account update occurs.
+                // this will update transaction details (like time, fee etc.)
+                dispatch(replaceTransactionThunk({ tx: precomposedTx, newTxid: txid }));
```
Comment (szymonlesisz) [inline packages/suite/src/actions/wallet/sendFormActions.ts:262]: not neccary, txid is available

File: packages/connect/src/api/bitcoin/createPending.ts (hunk 4 above)
Comment (mroz22) [reply packages/connect/src/api/bitcoin/createPending.ts]: [563582e](https://github.com/trezor/trezor-suite/pull/7783/commits/563582eddd132b0c119c25c0c96d336a22044346)

File: packages/connect/src/api/signTransaction.ts (hunk 3 above)
Comment (mroz22) [reply packages/connect/src/api/signTransaction.ts]: [563582e](https://github.com/trezor/trezor-suite/pull/7783/commits/563582eddd132b0c119c25c0c96d336a22044346)

File: packages/connect/src/api/bitcoin/index.ts (hunk 2 above)
Comment (mroz22) [reply packages/connect/src/api/bitcoin/index.ts]: [563582e](https://github.com/trezor/trezor-suite/pull/7783/commits/563582eddd132b0c119c25c0c96d336a22044346)

File: packages/connect/src/api/bitcoin/createPending.ts (hunk 1 above)
Comment (mroz22) [reply packages/connect/src/api/bitcoin/createPending.ts]: [563582e](https://github.com/trezor/trezor-suite/pull/7783/commits/563582eddd132b0c119c25c0c96d336a22044346)

File: packages/connect/src/api/signTransaction.ts (hunk 5 above)
Comment (mroz22) [reply packages/connect/src/api/signTransaction.ts]: [563582e](https://github.com/trezor/trezor-suite/pull/7783/commits/563582eddd132b0c119c25c0c96d336a22044346)

File: packages/suite/src/actions/wallet/sendFormActions.ts (hunk 6 above)
Comment (mroz22) [reply packages/suite/src/actions/wallet/sendFormActions.ts:262]: [e683c20](https://github.com/trezor/trezor-suite/pull/7783/commits/e683c20740e77d8e3dc5efd79dbb177aeaabb74e)

**Comment by mroz22** [comment]
resolved conflict in connect/changelog

**Review by szymonlesisz — APPROVED** [review]

**Pull request merged**